plugins/parsers/xpath/protocolbuffer_document.go

```diff
@@ -0,0 +1,140 @@
+package xpath
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/dynamicpb"
+
+	path "github.com/antchfx/xpath"
+	"github.com/doclambda/protobufquery"
+
+	// Register all known definitions
+	_ "github.com/doclambda/protobufquery/testcases/addressbook"
+	_ "github.com/doclambda/tahu/client_libraries/golang"
```

Comment:
When does the protobuf functionality require a user to add definitions here and recompile?

I haven't use dynamicpb before. Does it allow us to read an arbitrary protobuf message without a .proto/.pb.go and without recompiling?

Reply:
I tried really hard to make protobufs parseable by specifying the `.proto` file. However, there is currently no way to dynamically compile this definition to anything useable from within Go. Therefore, adding new protocol buffers require a PR with an added import as shown above for the `tahu` library.

Reply:
I don't think we're going to want to merge the protobuf parser if it can't handle arbitrary protobuf messages at runtime.

Users aren't going to want to submit a PR and wait for a release for every new protobuf message type they use. People often use protobufs for internal messages so they might not even want to publish their .proto. We're also not going to want to manage merging PRs for every new protobuf message type for all users.

It would make more sense to me to have a separate plugin for each protobuf-based protocol (like sparkplug_b) than have support here. Maybe these separate plugins could use the xpath code from this plugin instead.

Reply:
TBH I don't see the difference. Imagine you have someone wanting to get data from e.g. `sparkplug_b`. With the parser approach proposed here, you need to add one import to make a new protobuf type work. With your suggestion you will need a whole new plugin! I think the review effort is massively different between the two.
However, if this is your last word, then I will remove the protobuf part.

Reply:
Ok I found a way to dynamically load the proto file. No need for compile-time modifications anymore...